lti_placements/canvas_placements.py

```diff
@@ -0,0 +1,261 @@
+# Script to get all sites where Zoom is visible and retrieve the meetings to generate a report
+
+import json
+import logging
+import math
+import os
+import re
+from typing import Dict, List, Optional, Sequence, Union
+
+import canvasapi
+import pandas as pd
+import requests
+from bs4 import BeautifulSoup as bs
+
+from db.db_creator import DBCreator
+from environ import ENV, DATA_DIR
+from vocab import DataSourceStatus, ValidDataSourceName
+
+logger = logging.getLogger(__name__)
+
+# This should be uppercase lablels from the canvas tabs call
+# All of these will be included in the report
+# GET /api/v1/courses/:course_id/tabs
+# TODO: Make this configurable?
+SUPPORTED_PLACEMENTS = {"ZOOM": 0, "BLUEJEANS": 1}
```

Comment:
Yeah I was probably thinking that too, since the only place it's used and needed is this file, defining it outside probably wasn't worthwhile. The only reason to have it in vocab I'd think would be to make it easier to find as a reference for what ID = what type. I don't think it matters that much and maybe that can be changed if this changes in the future.

Reply:
This is probably a moot point by now.  Consider it a thought experiment, if you'd like.

I like @ssciolla's suggestion to use an enum.  You've probably heard my rants about avoiding strings in code that should be symbols and the ambiguity of dictionaries.  So, to address the issues @jonespm  mentioned, I thought…

> I left it a dict because it has to do an `in` check and the syntax for retrieving keys was better than an enum.

If the enum were `PlacementType`, you could get a list of its elements like this…

```python
class PlacementType(IntEnum):
    ZOOM = 0
    BLUEJEANS = 1

placementNames = list(map(lambda p: p.name, PlacementType))
```

Then you could use `placementNames` for the `in` comparison.  You could add this code to the `PlacementType` class as a property that you refer to when needed.  (E.g., `PlacementType.names`)

> I also thought it might need more than just the value, it might need to be something like
> 
> `"ZOOM": {"id": 0, "detail_function": "get_zoom_details"}`
> 
> This is kind of similar to what you have with ValidJobName but I'm going to need a little more configuration for some of these. I also don't know if this ever needs to be configurable. I could move it back to vocab though.

Add a `get_details` method to `PlacementType` that would do the required work based on the value or would make a call to the appropriate Zoom or BlueJeans code based on the value.